tcms/testruns/models.py

```diff
@@ -256,9 +257,21 @@ class TestExecution(TCMSActionModel):
         on_delete=models.CASCADE,
     )
     case_text_version = models.IntegerField()
-    close_date = models.DateTimeField(null=True, blank=True)
+    start_date = models.DateTimeField(null=True, blank=True, db_index=True)
+    stop_date = models.DateTimeField(null=True, blank=True, db_index=True)
     sortkey = models.IntegerField(null=True, blank=True)
 
+    def _actual_duration(self):
+        duration_difference = self.start_date - self.stop_date
+        return duration_difference
+
+    actual_duration = property(_actual_duration)
+
+    def _expected_duration(self):
+        return self.case.expected_duration
+
+    expected_duration = property(_expected_duration)
```

Comment:
This is blocked on #2253 b/c at the moment `case.expected_duration` does not exist.